eis_toolkit/exploratory_analyses/k_means_cluster.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import KMeans
+from sklearn.preprocessing import StandardScaler
+
+from eis_toolkit.exceptions import EmptyDataFrameException
+
+
+def _k_means_clustering(
+    data: gdp.GeoDataFrame, number_of_clusters: int, random_state: Optional[int] = None
+) -> gdp.GeoDataFrame:
```

Comment:
No need to define default values in an inner function, so you can remove the "= None" at the end